### PR TITLE
fix: Outliers endpoint "orderBy" not respecting "headers" name

### PR DESCRIPTION
**Pull request by d-bernat**
The 'Outliers API' offers the 'OrderBy' query parameter for sorting database data in the SELECT SQL statement. However, the parameter did not adhere to the convention of having the same name as the header names.

### CODE REVIEW

File: dhis-2/dhis-api/src/main/java/org/hisp/dhis/feedback/ErrorCode.java [hunk 1]
```diff
@@ -157,7 +157,7 @@ public enum ErrorCode {
   E2210("Data end date not allowed"),
   E2211("Algorithm min-max values not allowed"),
   E2212("Specifying both a start date/end date and a relative period is not allowed"),
-
+  E2213("Order by value is not compatible with algorithm"),
```
Comment (gnespolino) [inline dhis-2/dhis-api/src/main/java/org/hisp/dhis/feedback/ErrorCode.java:160]: could we add `{0}` to return which algorithm ?

**Review by gnespolino — APPROVED** [review]

File: dhis-2/dhis-api/src/main/java/org/hisp/dhis/feedback/ErrorCode.java (hunk 1 above)
Comment (maikelarabori) [reply dhis-2/dhis-api/src/main/java/org/hisp/dhis/feedback/ErrorCode.java:160]:
```suggestion
  E2213("Value of param orderBy is not compatible with algorithm `{0}`"),
```

File: dhis-2/dhis-api/src/main/java/org/hisp/dhis/feedback/ErrorCode.java [hunk 2]
```diff
@@ -444,6 +444,7 @@ public enum ErrorCode {
 
   E7180(
       "The analytics outliers data does not exist. Please ensure analytics job was run and did not skip the outliers"),
+  E7181("Order by param `{0}` does not exist"),
```
Comment (maikelarabori) [inline dhis-2/dhis-api/src/main/java/org/hisp/dhis/feedback/ErrorCode.java:447]:
```suggestion
  E7181("Column `{0}` specified, in orderBy, does not exist"),
```

File: dhis-2/dhis-services/dhis-service-analytics/src/main/java/org/hisp/dhis/analytics/outlier/data/OutlierQueryParser.java [hunk 3]
```diff
@@ -164,4 +168,54 @@ private List<Period> getPeriods(RelativePeriodEnum relativePeriod, Date relative
...
         .toList();
   }
+
+  /**
+   * The method retrieves the enumerator item related to the column for order by statement.
+   *
+   * @param orderBy is the order by field. It has to be the same as header item name.
+   * @return the {@link Order}.
+   */
+  private Order getOrderBy(String orderBy) {
```
Comment (maikelarabori) [inline dhis-2/dhis-services/dhis-service-analytics/src/main/java/org/hisp/dhis/analytics/outlier/data/OutlierQueryParser.java:178]: We could, probably, replace all these "IFs" by creating a method inside the enum Order that returns the respective Order based on the `orderBy` (which should be renamed to `headerName`).

Comment (d-bernat) [reply dhis-2/dhis-services/dhis-service-analytics/src/main/java/org/hisp/dhis/analytics/outlier/data/OutlierQueryParser.java:178]: done

File: dhis-2/dhis-api/src/main/java/org/hisp/dhis/feedback/ErrorCode.java (hunk 2 above)
Comment (d-bernat) [reply dhis-2/dhis-api/src/main/java/org/hisp/dhis/feedback/ErrorCode.java:447]: not all columns are eligible for order by and still can exist. What about Column `{0}` specified, in orderBy, is not eligible for orderBy or does not exist

File: dhis-2/dhis-api/src/main/java/org/hisp/dhis/feedback/ErrorCode.java (hunk 1 above)
Comment (d-bernat) [reply dhis-2/dhis-api/src/main/java/org/hisp/dhis/feedback/ErrorCode.java:160]: done

File: dhis-2/dhis-api/src/main/java/org/hisp/dhis/feedback/ErrorCode.java (hunk 2 above)
Comment (maikelarabori) [reply dhis-2/dhis-api/src/main/java/org/hisp/dhis/feedback/ErrorCode.java]: Makes sense @d-bernat 👍

**Review by maikelarabori — APPROVED** [review]

**Pull request merged**